packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -319,6 +344,33 @@ contract RoutersFacet is BaseConnextFacet {
     emit LiquidityFeeNumeratorUpdated(_numerator, msg.sender);
   }
 
+  /**
+   * @notice Allow router to use Portals
+   * @param _router - The router address to approve
+   */
+  function approveRouterForPortal(address _router) external {
+    if (!s.routerPermissionInfo.approvedRouters[_router]) revert RoutersFacet__approveRouterForPortal_notRouter();
+    if (s.routerPermissionInfo.approvedForPortalRouters[_router])
+      revert RoutersFacet__approveRouterForPortal_alreadyApproved();
+
+    s.routerPermissionInfo.approvedForPortalRouters[_router] = true;
+
+    emit RouterApprovedForPortal(_router, msg.sender);
+  }
+
+  /**
+   * @notice Remove router access to use Portals
+   * @param _router - The router address to remove approval
+   */
+  function disapproveRouterForPortal(address _router) external {
```

Comment:
1. nit: should be `unapproveRouterForPortal` I think?

2. doesn't necessarily *need* to be onlyOwner so long as it's onlyRouterOwner.... buuuut, 1 huge problem here: if router owes debt to aave portal, and they are unapproved/disapproved they will be unable to pay back that debt via `PortalFacet.repayAavePortal`. I imagine that has some pretty bad consequences.

So first, **permission this pls**

next, either one of two things must happen:
- Can't unapprove routers with outstanding debt (not a fan)
- Remove the check to make sure router is approved for portals in `PortalFacet.repayAavePortal` (am a fan)